### ISSUE TITLE
Export non-readable meshes at runtime

### ISSUE THREAD

**Issue by atteneder** [atteneder/glTFast#503]
Meshes that are not marked as readable cannot be exported at the moment (was reported, did not try myself)

The native buffers could be read nonetheless, but with a different API:

See [Rendering.AsyncGPUReadback.RequestIntoNativeArray](https://docs.unity3d.com/2023.1/Documentation/ScriptReference/Rendering.AsyncGPUReadback.RequestIntoNativeArray.html)

Sample code:

```csharp
var requestVertexBuf = AsyncGPUReadback.Request(mesh.sharedMesh.GetVertexBuffer(0));
var requestIndexBuf = AsyncGPUReadback.Request(mesh.sharedMesh.GetIndexBuffer());

requestVertexBuf.WaitForCompletion();
requestIndexBuf.WaitForCompletion();

var nativeVertices = requestVertexBuf.GetData<Vector3>();
var nativeIndices = requestIndexBuf.GetData<ushort>();
```

We got to investigate if this works and implement it properly.

**Comment by andreasherbig** [comment]
Just stumbled upon this exact issue. Would be great if the different approach worked...

**Comment by andreasherbig** [comment]
Oh, maybe this stacktrace will save a couple of minutes:
```
InvalidOperationException: Not allowed to access vertex data on mesh 'Cube' at array index 0 (isReadable is false; Read/Write must be enabled in import settings)
UnityEngine.Mesh+MeshDataArray..ctor (UnityEngine.Mesh[] meshes, System.Int32 meshesCount, System.Boolean checkReadWrite) (at <ad50157ee00e45cdb3c8bd67012f8804>:0)
UnityEngine.Mesh.AcquireReadOnlyMeshData (System.Collections.Generic.List`1[T] meshes) (at <ad50157ee00e45cdb3c8bd67012f8804>:0)
GLTFast.Export.GltfWriter.BakeMeshes () (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Runtime/Scripts/Export/GltfWriter.cs:622)
GLTFast.Export.GltfWriter.Bake (System.String bufferPath, System.String directory) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Runtime/Scripts/Export/GltfWriter.cs:499)
GLTFast.Export.GltfWriter.SaveAndDispose (System.IO.Stream outStream, System.String bufferPath, System.String directory) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Runtime/Scripts/Export/GltfWriter.cs:336)
GLTFast.Export.GltfWriter.SaveToFileAndDispose (System.String path) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Runtime/Scripts/Export/GltfWriter.cs:303)
GLTFast.Export.GameObjectExport.SaveToFileAndDispose (System.String path) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Runtime/Scripts/Export/GameObjectExport.cs:93)
AsyncHelpers+<>c__DisplayClass1_0`1[T].<RunSync>b__0 (System.Object _) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/AsyncHelpers.cs:61)
Rethrow as AggregateException: AsyncHelpers.Run method threw an exception. (Not allowed to access vertex data on mesh 'Cube' at array index 0 (isReadable is false; Read/Write must be enabled in import settings))
AsyncHelpers+ExclusiveSynchronizationContext.BeginMessageLoop () (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/AsyncHelpers.cs:117)
AsyncHelpers.RunSync[T] (System.Func`1[TResult] task) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/AsyncHelpers.cs:68)
GLTFast.Editor.MenuEntries.Export (System.Boolean binary, System.String name, UnityEngine.GameObject[] gameObjects) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/MenuEntries.cs:118)
GLTFast.Editor.MenuEntries.ExportGameObject (UnityEditor.MenuCommand command, System.Boolean binary) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/MenuEntries.cs:88)
GLTFast.Editor.MenuEntries.ExportGameObjectBinaryMenu (UnityEditor.MenuCommand command) (at Library/PackageCache/com.atteneder.gltfast@04962b4761/Editor/Scripts/MenuEntries.cs:82)
UnityEditor.GenericMenu:CatchMenu(Object, String[], Int32)
```

**Comment by andreasherbig** [comment]
KhronosGroups' UnityGLTF is working around this issue by promting the user to change the readable property of the mesh for the import:


```
#if UNITY_EDITOR
			if (!MeshIsReadable(meshObj) && EditorUtility.IsPersistent(meshObj))
			{
#if UNITY_2019_3_OR_NEWER
				if(EditorUtility.DisplayDialog("Exporting mesh but mesh is not readable",
					   $"The mesh {meshObj.name} is not readable. Do you want to change its import settings and make it readable now?",
					   "Make it readable", "No, skip mesh",
					   DialogOptOutDecisionType.ForThisSession, MakeMeshReadableDialogueDecisionKey))
#endif
				{
					var path = AssetDatabase.GetAssetPath(meshObj);
					var importer = AssetImporter.GetAtPath(path) as ModelImporter;
					if (importer)
					{
						importer.isReadable = true;
						importer.SaveAndReimport();
					}
				}
#if UNITY_2019_3_OR_NEWER
				else
				{
					Debug.LogWarning($"The mesh {meshObj.name} is not readable. Skipping");
					return null;
				}
#endif
			}
#endif
```

**Comment by atteneder** [comment]
That would be an improvement that's not too hard to implement. Would you like to champion this change?

**Comment by andreasherbig** [comment]
I'll see if I can spare the time :) Will keep you posted.

**Comment by atteneder** [comment]
Released in glTFast [6.8.0](https://docs.unity3d.com/Packages/com.unity.cloud.gltfast@6.8/changelog/CHANGELOG.html#680---2024-09-05).
